pkg/lib/registry/registry.go

```diff
@@ -354,3 +373,40 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 
 	return nil
 }
+
+// CheckForBundlePaths verifies presence of a list of bundle paths in the registry.
+func CheckForBundlePaths(querier registry.Query, bundlePaths []string) ([]string, []string, error) {
+	if len(bundlePaths) == 0 {
+		return bundlePaths, nil, nil
+	}
+
+	registryBundles, err := querier.ListBundles(context.TODO())
+	if err != nil {
+		return bundlePaths, nil, err
+	}
+
+	if len(registryBundles) == 0 {
+		return nil, bundlePaths, nil
+	}
+
+	registryBundlePaths := make(map[string]struct{})
+	for _, b := range registryBundles {
+		registryBundlePaths[b.BundlePath] = struct{}{}
+	}
+
+	var found, missing []string
+	for _, b := range bundlePaths {
+		if _, ok := registryBundlePaths[b]; ok {
+			if found == nil {
+				found = make([]string, 0)
+			}
+			found = append(found, b)
+			continue
+		}
+		if missing == nil {
+			missing = make([]string, 0)
+		}
```

Comment:
this isn't necessary -- `append` works with `nil` field values so long as the field is declared as a slice type.